src/source/noise.rs

```diff
@@ -0,0 +1,135 @@
+use crate::Source;
+
+use super::SeekError;
+
+/// Create a new white noise source.
+#[inline]
+pub fn white(sample_rate: cpal::SampleRate) -> WhiteNoise {
+    WhiteNoise::new(sample_rate)
+}
+
+/// Create a new pink noise source.
+#[inline]
+pub fn pink(sample_rate: cpal::SampleRate) -> PinkNoise {
+    PinkNoise::new(sample_rate)
+}
+
+/// Generates an infinite stream of random samples in [=1.0, 1.0]
+#[derive(Clone, Debug)]
+pub struct WhiteNoise {
+    sample_rate: cpal::SampleRate,
+}
+
+impl WhiteNoise {
+    /// Create a new white noise generator.
+    pub fn new(sample_rate: cpal::SampleRate) -> Self {
+        Self { sample_rate }
+    }
+}
+
+impl Iterator for WhiteNoise {
+    type Item = f32;
+
+    #[inline]
+    fn next(&mut self) -> Option<Self::Item> {
+        let randf = rand::random::<f32>();
```

Comment:
take a look at the random number generators in the rand crate. Most of them are very slow but in exchange output cryptographically secure numbers. I do not think that is needed here but we do pay the price for it.

Reply:
Will do. I'm using the `cpal::SampleRate` type for sample rate parameter where we ask for it. Is that okay or would you rather it's just a `u32`?

Reply:
In a015eb49a2

Reply:
> Will do. I'm using the `cpal::SampleRate` type for sample rate parameter where we ask for it. Is that okay or would you rather it's just a `u32`?

I would match what the other signal generator/synthesizer sources do (Sine is one I believe?)